File: 3301.js
function getSobrinhoMeio(h, z, l) {
  if ((h > z && h < l) || (h < z && h > l)) {
    return 'huguinho';
  }

  if ((z > h && z < l) || (z < h && z > l)) {
    return 'zezinho';
  }

  return 'luisinho';
}

function main() {
  var input = require('fs').readFileSync('/dev/stdin', 'utf8');
  var lines = input.split('\n');

  const [h, z, l] = lines.shift().split(' ').map(n => Number(n));

  console.log(getSobrinhoMeio(h, z, l));
}

main();

File: 3301.py
def get_sobrinho_meio(h, z, l):
  if (h > z and h < l) or (h < z and h > l):
    return 'huguinho'

  elif (z > h and z < l) or (z < h and z > l):
    return 'zezinho'

  return 'luisinho'

def main():
  h, z, l = map(int, input().split())

  print(get_sobrinho_meio(h, z, l))

if __name__ == '__main__':
  main()